### PR TITLE
Merge hidden main

### PR DESCRIPTION
**Pull request by ShadoySV**

### CODE REVIEW

File: contracts/tokenomics/xastro_token/src/contract.rs [hunk 1]
```diff
@@ -255,7 +255,7 @@ pub fn execute(
             marketing,
         } => execute_update_marketing(deps, env, info, project, description, marketing),
         ExecuteMsg::UploadLogo(logo) => execute_upload_logo(deps, env, info, logo),
-        _ => Err(ContractError::Unauthorized {}),
+        _ => Err(StdError::generic_err("Unsupported execute message").into()),
```
Comment (epanchee) [inline contracts/tokenomics/xastro_token/src/contract.rs:258]: Need to update version and add empty migration for this contract

**Review by P-Yevhenii — APPROVED** [review]

File: contracts/pair/src/contract.rs [hunk 2]
```diff
@@ -9,8 +9,8 @@ use cosmwasm_std::{
 
 use crate::response::MsgInstantiateContractResponse;
 use astroport::asset::{
-    addr_opt_validate, addr_validate_to_lower, check_swap_parameters, format_lp_token_name, Asset,
-    AssetInfo, PairInfo, MINIMUM_LIQUIDITY_AMOUNT,
+    addr_opt_validate, check_swap_parameters, format_lp_token_name, Asset, AssetInfo, CoinsExt,
```
[the remark this answers is not in the document]
Comment (ShadoySV) [reply contracts/pair/src/contract.rs:12]: Will do on the next merge step with release branch

**Review by epanchee — APPROVED** [review]

File: contracts/pair_astro_xastro/src/lib.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@ use crate::contract::Contract;
 pub mod contract;
 pub mod state;
 
-use crate::state::MigrateMsg;
+use crate::state::{InitParams, MigrateMsg};
```
[the remark this answers is not in the document]
Comment (ShadoySV) [reply contracts/pair_astro_xastro/src/lib.rs:6]: Will do on the next merge step with release branch

File: contracts/pair_concentrated/Cargo.toml [hunk 4]
```diff
@@ -39,4 +39,6 @@ cw-multi-test = "0.15"
 astroport-factory = { path = "../factory" }
 proptest = "1.0"
 anyhow = "1.0"
-derivative = "2.2"
\ No newline at end of file
+derivative = "2.2"
```
[the remark this answers is not in the document]
Comment (ShadoySV) [reply contracts/pair_concentrated/Cargo.toml:42]: Will do on the next merge step with release branch

**Pull request merged**